components/authz-service/server/v2/system_integration_test.go

```diff
@@ -57,6 +59,103 @@ func TestIntegrationSystemPolicies(t *testing.T) {
+	ts := setupWithOPAV2p1(t)
+	defer ts.Shutdown(t, ctx)
+	cl := ts.Authz
+
+	user := "user:local:alice"
+	authorizedProjectId := "authorized-project"
+	unauthorizedProjectId := "project-not-authorized"
+
+	_, err := ts.Projects.CreateProject(ctx, &api_v2.CreateProjectReq{
+		Id:   authorizedProjectId,
+		Name: "Project Authorized",
+	})
+	require.NoError(t, err)
+
+	_, err = ts.Projects.CreateProject(ctx, &api_v2.CreateProjectReq{
+		Id:   unauthorizedProjectId,
+		Name: "Project Unauthorized",
+	})
+	require.NoError(t, err)
+
+	statement := api_v2.Statement{
+		Effect:    api_v2.Statement_ALLOW,
+		Resources: []string{"*"},
+		Actions:   []string{"iam:projects:assign"},
+		Projects:  []string{authorizedProjectId},
+	}
+	req := api_v2.CreatePolicyReq{
+		Id:         "policy-2",
+		Name:       "my favorite policy",
+		Members:    []string{user},
+		Statements: []*api_v2.Statement{&statement},
+	}
+	_, err = ts.Policy.CreatePolicy(ctx, &req)
+	require.NoError(t, err)
+
+	// force sync refresh
+	err = ts.PolicyRefresher.Refresh(ctx)
+	require.NoError(t, err)
+
+	cases := map[string]func(*testing.T){
+		"when passed one unauthorized project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{unauthorizedProjectId},
+			})
+			grpctest.AssertCode(t, codes.PermissionDenied, err)
+		},
+		"when passed one not found project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{"project-not-found"},
+			})
+			grpctest.AssertCode(t, codes.NotFound, err)
+		},
+		"when passed one authorized and one not found project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{authorizedProjectId, "project-not-found"},
+			})
+			grpctest.AssertCode(t, codes.NotFound, err)
+		},
+		"when passed one unauthorized project and one not found project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{unauthorizedProjectId, "project-not-found"},
+			})
+			grpctest.AssertCode(t, codes.NotFound, err)
+		},
+		"when passed one unauthorized project and one authorized project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{unauthorizedProjectId, authorizedProjectId},
+			})
+			grpctest.AssertCode(t, codes.PermissionDenied, err)
+		},
+		"when passed one authorized project": func(t *testing.T) {
+			_, err := cl.ValidateProjectAssignment(ctx, &api_v2.ValidateProjectAssignmentReq{
+				Subjects:   []string{user},
+				ProjectIds: []string{authorizedProjectId},
+			})
+			assert.NoError(t, err)
+		},
```

Comment:
Consider adding a test with two unauthorized projects and another with two non-existent projects. Then, when you check the error text, too (as mentioned above).
Those will answer this question: does the error message mention all such projects or just the first one?